greenplumpython/func.py

```diff
@@ -170,292 +158,325 @@ def serialize(self) -> str:
-
-    return make_function_call
+    if wrapped_func is None:
+        return functools.partial(create_function, language_handler=language_handler)
+    return NormalFunction(wrapped_func=wrapped_func, language_handler=language_handler)
 
 
 # FIXME: Add test cases for optional parameters
 def create_aggregate(
-    trans_func: Optional[Callable] = None,  # type: ignore
-    name: Optional[str] = None,
-    schema: Optional[str] = None,
-    temp: bool = True,
-    language_handler: str = "plpython3u",
-) -> Callable:  # type: ignore
+    transition_func: Optional[Callable[..., Any]] = None, language_handler: str = "plpython3u"
+) -> AggregateFunction:
     """
-    Creates a User Defined Python Aggregation (UDA) in Greenplum Database according to the Python
-    function code given.
+    Creates a User Defined Aggregate (UDA) in Database using the given Python
+    function as the state transition function.
 
     Args:
-        trans_func : python function
-        name : Optional[str] : name of the UDA
-        schema : Optional[str] : where the UDA will be stored
-        temp: bool : if the creation of the UDA is temporary exists only for current session
-        language_handler : str : language handler extension to create UDA in Greenplum, by default plpython3u, but can also choose plcontainer.
+        transition_func : python function
+        language_handler : language handler to run the aggregate function in database,
+            defaults to plpython3u, will also support plcontainer later.
 
     Returns:
-        Callable : FunctionCall
+        A database aggregate function.
 
     Example:
-            .. code-block::  Python
+        .. code-block::  Python
 
-                @gp.create_aggregate
-                def my_sum(result: int, val: int) -> int:
-                    if result is None:
-                        return val
-                    return result + val
+            @gp.create_aggregate
+            def my_sum(result: int, val: int) -> int:
+                if result is None:
+                    return val
+                return result + val
 
-                rows = [(1,) for _ in range(10)]
-                numbers = gp.values(rows, db=db, column_names=["val"])
-                results = list(my_sum(numbers["val"], as_name="result").to_table().fetch())
+            rows = [(1,) for _ in range(10)]
+            numbers = gp.values(rows, db=db, column_names=["val"])
+            my_sum(numbers["val"])
 
     """
     # If user needs extra parameters when creating a function
-    if not trans_func:
-        return functools.partial(
-            create_aggregate,  # type: ignore
-            name=name,
-            schema=schema,
-            temp=temp,
+    if transition_func is None:
+        return functools.partial(create_aggregate, language_handler=language_handler)
+    return AggregateFunction(
+        transition_func=NormalFunction(
+            transition_func,
+            name="func_" + uuid4().hex,
```

Comment:
not for this PR scope, do we consider to use a shorter uuid? like 
https://github.com/skorokithakis/shortuuid

Reply:
```
>>> shortuuid.uuid()
b'vytxeTZskVKR7C7WgdSP3d'

>>> base64.standard_b64encode(uuid4().bytes)
b'7tKLaUQoTR+PDeOszhDWCg=='

>>> uuid4().hex
b'a75f590faaa64d70b19a5f2cc9bf0778'
```

8 bytes shorter? Anyway, not an issue related to this PR